miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -215,23 +212,30 @@ public Sample get(Long sampleId) throws IOException {
     authorizationManager.throwIfNotReadable(sample);
     return sample;
   }
-  
+
   @Override
   public Long create(Sample sample) throws IOException {
     loadChildEntities(sample);
     authorizationManager.throwIfNotWritable(sample);
     setChangeDetails(sample, true);
-    if (sample.getSampleAdditionalInfo() != null) { 
-      if (sample.getSampleAdditionalInfo().getSampleClass() == null 
+    if (sample.getSampleAdditionalInfo() != null) {
+      if (sample.getSampleAdditionalInfo().getSampleClass() == null
           || sample.getSampleAdditionalInfo().getSampleClass().getSampleCategory() == null) {
         throw new IllegalArgumentException("Sample class or category missing");
       }
       if (!Identity.CATEGORY_NAME.equals(sample.getSampleAdditionalInfo().getSampleClass().getSampleCategory())) {
         sample.getSampleAdditionalInfo().setParent(findOrCreateParent(sample));
         validateHierarchy(sample);
+        if (sample.getSampleAdditionalInfo().getGroupId() == null) {
+          sample.getSampleAdditionalInfo().setGroupId(sample.getSampleAdditionalInfo().getParent().getSampleAdditionalInfo().getGroupId());
+        }
+        if (sample.getSampleAdditionalInfo().getGroupDescription() == null) {
+          sample.getSampleAdditionalInfo()
+              .setGroupDescription(sample.getSampleAdditionalInfo().getParent().getSampleAdditionalInfo().getGroupDescription());
+        }
```

Comment:
So if the parent has a group ID, it is ok to set the child's group ID to something different, and this will be saved, but if you clear the group ID, it will get changed back to the parent's group ID in the save process, likely without the user noticing? Then when editing the sample, it's fine to clear it, and that will be saved?